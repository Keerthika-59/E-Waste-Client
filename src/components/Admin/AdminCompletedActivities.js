import React, { useEffect, useState, useMemo } from "react";
// import Header from "./Header";
import APIHelper from '../API/apihelper2'
import { TableHeader, Pagination, Search } from "./DashboardPages/Tablecomponent";
import useFullPageLoader from "./DashboardPages/useFullPageLoader";
// import AppConfig from "App.config";
import Swal from 'sweetalert2'
import { Button ,Table} from 'react-bootstrap'
import axios from 'axios'


const AdminCompletedActivities = () => {
    const [data, setReps] = useState([]);
    const [loader, showLoader, hideLoader] = useFullPageLoader();
    const [totalItems, setTotalItems] = useState(0);
    const [currentPage, setCurrentPage] = useState(1);
    const [search, setSearch] = useState("");
    const [sorting, setSorting] = useState({ field: "", order: "" });

    const ITEMS_PER_PAGE = 9;

    const headers = [
        { name: "Name", field: "name", sortable: true },
        { name: "Email", field: "email", sortable: true },
        { name: "Phone Number", field: "phonenumber", sortable: false },
        { name: "City", field: "city", sortable: true },
        { name: "Address", field: "address", sortable: true },
        { name: "Action", field: "body", sortable: false }

    ];

    const url = 'https://ewaste-dec20-dev-api.azurewebsites.net/'

    const urls='http://localhost:5000/activities/completed'

    useEffect(() => {
        const getData = () => {
            showLoader();

            fetch(`${urls}`)
                .then(response => response.json())
                .then(json => {
                    hideLoader();
                    setReps(json);
                    console.log(json)
                });
        };

        getData();
    }, [data]);

    const repsData = useMemo(() => {
        let computedComments = data;

        if (search) {
            computedComments = computedComments.filter(
                rep =>
                    rep.name.toLowerCase().includes(search.toLowerCase()) ||
                    rep.email.toLowerCase().includes(search.toLowerCase()) ||
                    rep.city.toLowerCase().includes(search.toLowerCase()) ||
                    rep.address.toLowerCase().includes(search.toLowerCase())

            );
        }

        setTotalItems(computedComments.length);

        //Sorting comments
        if (sorting.field) {
            const reversed = sorting.order === "asc" ? 1 : -1;
            computedComments = computedComments.sort(
                (a, b) =>
                    reversed * a[sorting.field].localeCompare(b[sorting.field])
            );
        }

        //Current Page slice
        return computedComments.slice(
            (currentPage - 1) * ITEMS_PER_PAGE,
            (currentPage - 1) * ITEMS_PER_PAGE + ITEMS_PER_PAGE
        );
    }, [data, currentPage, search, sorting]);

    return (
        <>
            {/* <ExternalInfo page="datatable" /> */}

            <div className="row w-100">
                <div className="col mb-3 col-12 text-center">
                    <div className="row">
                        <div className="col-md-6">
                            <Pagination
                                total={totalItems}
                                itemsPerPage={ITEMS_PER_PAGE}
                                currentPage={currentPage}
                                onPageChange={page => setCurrentPage(page)}
                            />
                        </div>
                    </div>
                    {/* <div className="col-md-6 d-flex flex-row-reverse" > */}
                        {/* <Search
                            onSearch={value => {
                                setSearch(value);
                                setCurrentPage(1);
                            }}
                        /> */}
                    {/* </div> */}
                    <h4>Completed Activities</h4>
                    <Table responsive>
                        <thead>
                            <tr>
                                <th>Activity Id</th>
                                <th>User Name</th>
                                <th>Bio Waste</th>
                                <th>Non-Bio Waste</th>
                                <th>Donation</th>
                                <th>Representative Name</th>
                            </tr>
                        </thead>
                        <tbody>
                            {repsData.map(data => (
                                <tr>
                                    <td>{data._id}</td>
                                    <td>{data.userDetails.userName}</td>
                                    <td>{data.bioWaste ? 'Yes':"No"}</td>
                                    <td>{data.nonBioWaste ? 'Yes':'No'}</td>
                                    <td>{data.donation ? 'Yes':'No'}</td>
                                    { data.repDetails && <td>{data.repDetails.repName}</td>}

                                    {/* <td>{rep.repDetails.repId}</td> */}
                                </tr>
                            ))}
                        </tbody>
                    </Table>
                </div>
            </div>
        </>
    );
};

export default AdminCompletedActivities;
