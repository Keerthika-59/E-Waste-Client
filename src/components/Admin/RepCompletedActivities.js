// import React, { useEffect, useState, useMemo } from "react";
// // import Header from "./Header";
// import { TableHeader, Pagination, Search } from "./DashboardPages/Tablecomponent";
// import useFullPageLoader from "./DashboardPages/useFullPageLoader";
// // import AppConfig from "App.config";
// import Swal from 'sweetalert2'
// import { Button, Table } from 'react-bootstrap'

// const RepCompletedActivities = () => {
//     const [compActivity, setCompActivity] = useState([]);
//     const [loader, showLoader, hideLoader] = useFullPageLoader();
//     const [totalItems, setTotalItems] = useState(0);
//     const [currentPage, setCurrentPage] = useState(1);
//     const [search, setSearch] = useState("");
//     const [sorting, setSorting] = useState({ field: "", order: "" });

//     const ITEMS_PER_PAGE = 5;

//     const headers = [
//         { name: "Activity Id", field: "ActivityId", sortable: false },
//         { name: "Bio", field: "bio", sortable: false },
//         { name: "Non-Bio", field: "nonBio", sortable: false },
//         { name: "Donation", field: "donation", sortable: true },
//         { name: "User Id", field: "userId", sortable: false }

//     ];

//     useEffect(() => {
//         const getData = () => {
//             showLoader();

//             fetch("https://ewaste-dec20-dev-api.azurewebsites.net/reps")
//                 .then(response => response.json())
//                 .then(json => {
//                     hideLoader();
//                     setCompActivity(json);
//                 });
//         };

//         getData();
//     }, [compActivity]);

//     const UserCompActData = useMemo(() => {
//         let compAct = compActivity;

//         if (search) {
//             compAct = compAct.filter(
//                 act =>
//                     act.donation.toLowerCase().includes(search.toLowerCase())
//             );
//         }

//         setTotalItems(compAct.length);

//         //Sorting comments
//         if (sorting.field) {
//             const reversed = sorting.order === "asc" ? 1 : -1;
//             compAct = compAct.sort(
//                 (a, b) =>
//                     reversed * a[sorting.field].localeCompare(b[sorting.field])
//             );
//         }

//         //Current Page slice
//         return compAct.slice(
//             (currentPage - 1) * ITEMS_PER_PAGE,
//             (currentPage - 1) * ITEMS_PER_PAGE + ITEMS_PER_PAGE
//         );
//     }, [compActivity, currentPage, search, sorting]);

//     return (

//             <>
//             {/* <ExternalInfo page="datatable" /> */}

//             <div className="row w-100">
//                 <div className="col mb-3 col-12 text-center">
//                     <div className="row">
//                         <div className="col-md-6">
//                             <Pagination
//                                 total={totalItems}
//                                 itemsPerPage={ITEMS_PER_PAGE}
//                                 currentPage={currentPage}
//                                 onPageChange={page => setCurrentPage(page)}
//                             />
//                         </div> 

//                         {/* <Search

//                             onSearch={value => {
//                                 setSearch(value);
//                                 setCurrentPage(1);
//                             }}
//                         /> */}
//                     </div>
//                     <Table responsive >
//                         <thead>
//                             <tr>
//                                 <th>Activity Id</th>
//                                 <th>Bio</th>
//                                 <th>Non-bio</th>
//                                 <th>Donation</th>
//                                 <th>UserId</th>
//                             </tr>
//                         </thead>
//                     <tbody>
//                         {UserCompActData.map(act => (
//                             <tr>

//                                 <td>{act._id}</td>
//                                 <td>{act.bioWaste}</td>
//                                 <td>{act.nonBioWaste}</td>
//                                 <td>{act.donation}</td>
//                                 <td>{act.userId}</td>

//                             </tr>
//                         ))}
//                     </tbody>
//                     </Table>
//                 </div>
//                 <div className="col-md-6">
//                     <Pagination
//                         total={totalItems}
//                         itemsPerPage={ITEMS_PER_PAGE}
//                         currentPage={currentPage}
//                         onPageChange={page => setCurrentPage(page)}
//                     />
//                 </div>
//             </div>
//         </>
//     );
// };

// export default RepCompletedActivities;


import React, { useEffect, useState, useMemo } from "react";
import Cookies from "js-cookie";
import APIHelper from "../API/apihelper2";
import { Table } from "react-bootstrap";
import axios from "axios";


const RepCompletedActivities = () => {
  const [compActivity, setCompActivity] = useState([]);
  const [id, setId] = useState("");

  const fetchId = async () => {
    try {
      const token = Cookies.get("repr");
      const response = await APIHelper.fetchRepId(token)
      setId(response.data);

    } catch (e) {
      console.log(e);
    }
  };
    
//   useEffect(() => {
    
//   }, []);

  useEffect(() => {
    fetchId();
    const fetchReprData = async () => {
      try {

        const repData = await axios.get(`https://ewaste-dec20-dev-api.azurewebsites.net/representative/completed/605e1e67e5318a3b94a75a26`);
        console.log(repData.data.user_activities);

        setCompActivity(repData.data.user_activities);
      } catch (err) {
        console.log(err.response || err);
      }
    };
    fetchReprData();
  }, []);

  return (
    <>
      <Table responsive striped bordered hover>
        <thead>
          <tr>
            <th>Activity Id</th>
            <th>Bio Waste</th>
            <th>Non Bio Waste</th>
            <th>Donation</th>
            <th>User Id</th>
          </tr>
        </thead>
        <tbody>
            {/* {
                JSON.stringify(compActivity)
            } */}
          {compActivity.map((act) => (
              <tr>
                <td>{act._id}</td>
                <td>{act.bioWaste ? "Yes" : "No"}</td>
                <td>{act.nonBioWaste ? "Yes" : "No"}</td>
                <td>{act.donation ? "Yes" : "No"}</td>
                <td>{act.userDetails.userId}</td>
              </tr>
            ))}
        </tbody>
      </Table>
    </>
  );
};

export default RepCompletedActivities;