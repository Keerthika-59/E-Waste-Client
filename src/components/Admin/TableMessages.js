import React, { useEffect, useState, useMemo } from "react";
// import Header from "./Header";
import { TableHeader, Pagination, Search } from "./DashboardPages/Tablecomponent";
import useFullPageLoader from "./DashboardPages/useFullPageLoader";
// import AppConfig from "App.config";
import Swal from 'sweetalert2'
import { Button,Table } from 'react-bootstrap'
import axios from 'axios'

const TableMessages = () => {
    const [comments, setComments] = useState([]);
    const [loader, showLoader, hideLoader] = useFullPageLoader();
    const [totalItems, setTotalItems] = useState(0);
    const [currentPage, setCurrentPage] = useState(1);
    const [search, setSearch] = useState("");
    const [sorting, setSorting] = useState({ field: "", order: "" });

    const ITEMS_PER_PAGE = 15;

    const headers = [
        { name: "Name", field: "name", sortable: true },
        { name: "Email", field: "email", sortable: true },
        { name: "Message", field: "message", sortable: true },
        { name: "Time", field: "time", sortable: false }

    ];

    const url = 'http://ewaste-dec20-dev-api.azurewebsites.net/'

    useEffect(() => {
        const getData = () => {
            showLoader();

            fetch('http://ewaste-dec20-dev-api.azurewebsites.net/contacts')
                .then(response => response.json())
                .then(json => {
                    hideLoader();
                    setComments(json);
                });
        };

        getData();
    }, [comments]);

    const commentsData = useMemo(() => {
        let computedComments = comments;

        if (search) {
            computedComments = computedComments.filter(
                comment =>
                    comment.name.toLowerCase().includes(search.toLowerCase()) ||
                    comment.email.toLowerCase().includes(search.toLowerCase()) ||
                    comment.createdAt.toLowerCase().includes(search.toLocaleLowerCase())
            );
        }

        setTotalItems(computedComments.length);

        //Sorting comments
        if (sorting.field) {
            const reversed = sorting.order === "asc" ? 1 : -1;
            computedComments = computedComments.sort(
                (a, b) =>
                    reversed * a[sorting.field].localeCompare(b[sorting.field])
            );
        }

        //Current Page slice
        return computedComments.slice(
            (currentPage - 1) * ITEMS_PER_PAGE,
            (currentPage - 1) * ITEMS_PER_PAGE + ITEMS_PER_PAGE
        );
    }, [comments, currentPage, search, sorting]);

    return (
        <>
            {/* <ExternalInfo page="datatable" /> */}

            <div className="row w-100">
                <div className="col mb-3 col-12 text-center">
                    <div className="row">
                        <div className="col-md-6">
                            <Pagination
                                total={totalItems}
                                itemsPerPage={ITEMS_PER_PAGE}
                                currentPage={currentPage}
                                onPageChange={page => setCurrentPage(page)}
                            />
                        </div>
                    </div>
                    {/* <div className="col-md-6 d-flex flex-row-reverse" style={{ marginLeft: "400px" }}> */}
                    <Search
                        onSearch={value => {
                            setSearch(value);
                            setCurrentPage(1);
                        }}
                    />
                    {/* </div> */}
                    <h4>Messages</h4>

                    <Table responsive>
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>E-mail</th>
                                <th>Message</th>
                                <th>Action</th>
                            </tr>
                        </thead>
                        <tbody>
                            {commentsData.map(comment => (
                                <tr>
                                    <td>{comment.name}</td>
                                    <td>{comment.email}</td>
                                    <td>{comment.message}</td>
                                    <td>   <Button variant="danger" onClick={() => {

                                        Swal.fire({
                                            title: 'Are you sure?',
                                            text: 'You will not be able to recover this User Details!',
                                            icon: 'warning',
                                            showCancelButton: true,
                                            confirmButtonText: 'Yes, delete it!',
                                            cancelButtonText: 'No, keep it'
                                        }).then((result) => {
                                            if (result.value) {
                                                Swal.fire(
                                                    'Deleted!',
                                                    'The User details has been deleted.',
                                                    // 'success'
                                                )
                                                axios.delete(`${url}/admin/user/${comment._id}`)
                                            } else if (result.dismiss === Swal.DismissReason.cancel) {
                                                Swal.fire(
                                                    'Cancelled',
                                                    'The User details is not deleted :)',
                                                    // 'error'
                                                )
                                            }
                                        })


                                    }}>DELETE</Button> </td>
                                </tr>
                            ))}
                        </tbody>
                    </Table>
                </div>
            </div>
        </>
    );
};

export default TableMessages;
