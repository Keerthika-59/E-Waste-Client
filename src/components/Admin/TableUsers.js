import React, { useEffect, useState, useMemo } from "react";
// import Header from "./Header";
import APIHelper from '../API/apihelper'
import { TableHeader, Pagination, Search } from "./DashboardPages/Tablecomponent";
import useFullPageLoader from "./DashboardPages/useFullPageLoader";
// import AppConfig from "App.config";
import Swal from 'sweetalert2'
import {Button} from 'react-bootstrap'

const TableUsers = () => {
    const [users, setUsers] = useState([]);
    const [loader, showLoader, hideLoader] = useFullPageLoader();
    const [totalItems, setTotalItems] = useState(0);
    const [currentPage, setCurrentPage] = useState(1);
    const [search, setSearch] = useState("");
    const [sorting, setSorting] = useState({ field: "", order: "" });

    const ITEMS_PER_PAGE = 9;

    const headers = [
        { name: "No#", field: "id", sortable: false },
        { name: "Name", field: "name", sortable: true },
        { name: "Email", field: "email", sortable: true },
        { name: "Phone Number", field: "body", sortable: false },
        { name: "City", field: "city", sortable: true },
        { name: "Address", field: "address", sortable: true },
        { name: "Action", field: "body", sortable: false }
    ];

    useEffect(() => {
        const getData = () => {
            showLoader();

            fetch(`${APIHelper.API_URL}`)
                .then(response => response.json())
                .then(json => {
                    hideLoader();
                    setUsers(json);
                });
        };

        getData();
    }, []);

    const usersData = useMemo(() => {
        let computedComments = users;

        if (search) {
            computedComments = computedComments.filter(
                user =>
                    user.name.toLowerCase().includes(search.toLowerCase()) ||
                    user.email.toLowerCase().includes(search.toLowerCase())
            );
        }

        setTotalItems(computedComments.length);

        //Sorting comments
        if (sorting.field) {
            const reversed = sorting.order === "asc" ? 1 : -1;
            computedComments = computedComments.sort(
                (a, b) =>
                    reversed * a[sorting.field].localeCompare(b[sorting.field])
            );
        }

        //Current Page slice
        return computedComments.slice(
            (currentPage - 1) * ITEMS_PER_PAGE,
            (currentPage - 1) * ITEMS_PER_PAGE + ITEMS_PER_PAGE
        );
    }, [users, currentPage, search, sorting]);

    return (
        <>
            {/* <ExternalInfo page="datatable" /> */}

            <div className="row w-100">
                <div className="col mb-3 col-12 text-center">
                    <div className="row">
                        <div className="col-md-6">
                            <Pagination
                                total={totalItems}
                                itemsPerPage={ITEMS_PER_PAGE}
                                currentPage={currentPage}
                                onPageChange={page => setCurrentPage(page)}
                            />
                        </div>
                        <div className="col-md-6 d-flex flex-row-reverse">
                            <Search
                                onSearch={value => {
                                    setSearch(value);
                                    setCurrentPage(1);
                                }}
                            />
                        </div>
                    </div>

                    <table className="table table-striped">
                        <TableHeader
                            headers={headers}
                            onSorting={(field, order) =>
                                setSorting({ field, order })
                            }
                        />
                        <tbody>
                            {usersData.map(user => (
                                <tr>
                                    <th scope="row" key={user.id}>
                                        {user.id}
                                    </th>
                                    <td>{user.name}</td>
                                    <td>{user.email}</td>
                                    <td>{user.phoneNumber}</td>
                                    <td>{user.city}</td>
                                    <td>{user.address}</td>
                                    <td>   <Button variant="danger" onClick={() => {

                                        Swal.fire({
                                            title: 'Are you sure?',
                                            text: 'You will not be able to recover this User Details!',
                                            icon: 'warning',
                                            showCancelButton: true,
                                            confirmButtonText: 'Yes, delete it!',
                                            cancelButtonText: 'No, keep it'
                                        }).then((result) => {
                                            if (result.value) {
                                                Swal.fire(
                                                    'Deleted!',
                                                    'The User details has been deleted.',
                                                    // 'success'
                                                )
                                            } else if (result.dismiss === Swal.DismissReason.cancel) {
                                                Swal.fire(
                                                    'Cancelled',
                                                    'The User details is not deleted :)',
                                                    // 'error'
                                                )
                                            }
                                        })


                                    }}>DELETE</Button> </td>
                                </tr>
                            ))}
                        </tbody>
                    </table>
                </div>
            </div>
            {loader}
        </>
    );
};

export default TableUsers;
