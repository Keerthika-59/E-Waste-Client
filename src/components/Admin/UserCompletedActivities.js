import React, { useEffect, useState, useMemo } from "react";
import Cookies from "js-cookie";
import APIHelper from "../API/apihelper";
import { Table } from "react-bootstrap";
const UserCompletedActivities = () => {
  const [compActivity, setCompActivity] = useState([]);
  const [id, setId] = useState("");

  const fetchId = async () => {
    try {
      const token = Cookies.get("user");
      const id = await APIHelper.fetchUserId({
        token: token,
      });
      setId(id);
    } catch (e) {
      console.log(e);
    }
  };

  useEffect(() => {
    fetchId();
  }, []);

  useEffect(() => {
    const fetchUserData = async () => {
      try {
        const userData = await APIHelper.fetchUserData(id);
        console.log(userData.activity);
        setCompActivity(userData.activity);
      } catch (err) {
        console.log(err.response || err);
      }
    };
    fetchUserData();
  }, [id]);

  return (
    <>
      <Table responsive striped bordered hover>
        <thead>
          <tr>
            <th>Activity Id</th>
            <th>Bio Waste</th>
            <th>Non Bio Waste</th>
            <th>Donation</th>
            <th>Representative Id</th>
          </tr>
        </thead>
        <tbody>
          {compActivity
            .filter((activity) => activity.status === true)
            .map((act) => (
              <tr>
                <td>{act._id}</td>
                <td>{act.bioWaste ? "Yes" : "No"}</td>
                <td>{act.nonBioWaste ? "Yes" : "No"}</td>
                <td>{act.donation ? "Yes" : "No"}</td>
                <td>{act.repId}</td>
              </tr>
            ))}
        </tbody>
      </Table>
    </>
  );
};

export default UserCompletedActivities;