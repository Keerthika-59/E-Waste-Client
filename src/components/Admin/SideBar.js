import React, { useState, useContext } from 'react';
import * as FaIcons from 'react-icons/fa';
import * as BsIcons from 'react-icons/bs';

import { IconContext } from 'react-icons';
import { Link, Redirect } from 'react-router-dom';
import Footer from '../Footer/Footer';
import "./style.css"

import TableRepresentatives from './TableRepresentatives';
import TableVerifiedRep from './TableVerifiedRep';
import TableMessages from './TableMessages';
import TableUsers from './TableUsers';
import AuthApi from '../../authAPI';
import Cookies from 'js-cookie';
import AdminPendingActivities from './AdminPendingActivities'
import AdminCompletedActivities from './AdminCompletedActivities'

const SideBar = () => {

    const Auth = useContext(AuthApi);

    const [sidebar, setSidebar] = useState(true);

    const [show, setShow] = useState(0);

    // const showSidebar = () => setSidebar(!sidebar);

    return (
        (Auth.auth) ? <div className="sidebar1">
            <IconContext.Provider value={{ color: '#fff' }}>
                {/* <div className='sidenav'>
                    <Link to='#' className='menu-bars'>
                        <FaIcons.FaBars onClick = {showSidebar} />
                    </Link>
                </div> */}
                <div className="row">
                    <div className="col-md-4 sn">

                        <nav className={sidebar ? 'sidenav-menuad active' : 'sidenav-menuad'}>
                            <ul className='sidenav-menu-items'>

                                <li className=" mx-3 py-2 px-2" style={{ listStyle: 'none' }}
                                    onClick={(e) => setShow(0)}
                                >

                                    <h5 style={{ color: 'white' }}>
                                        <BsIcons.BsFillPersonFill />  View Users
                                      </h5>
                                </li>

                                <li className="mx-3 py-2 px-2" style={{ listStyle: 'none' }}
                                    onClick={(e) => setShow(1)}
                                >

                                    <h5 style={{ color: 'white' }}>
                                        <BsIcons.BsFillPersonFill />  View Reps
                                      </h5>
                                </li>

                                <li className="mx-3 py-2 px-2" style={{ listStyle: 'none' }}
                                    onClick={(e) => setShow(2)}
                                >

                                    <h5 style={{ color: 'white' }}>
                                        <BsIcons.BsFillPersonFill />  View Messages
                                      </h5>
                                </li>

                                <li className="mx-3 py-2 px-2" style={{ listStyle: 'none' }}
                                    onClick={(e) => setShow(3)}
                                >

                                    <h5 style={{ color: 'white' }}>
                                        <BsIcons.BsFillPersonFill />  Verify Reps
                                      </h5>
                                </li>
                                <li className="mx-3 py-2 px-2" style={{ listStyle: 'none' }}
                                    onClick={(e) => setShow(4)}
                                >

                                    <h5 style={{ color: 'white' }}>
                                        <BsIcons.BsFillPersonFill />  Pending Activities
                                      </h5>
                                </li>
                                <li className="mx-3 py-2 px-2" style={{ listStyle: 'none' }}
                                    onClick={(e) => setShow(5)}
                                >

                                    <h5 style={{ color: 'white' }}>
                                        <BsIcons.BsFillPersonFill />  Completed Activities
                                      </h5>
                                </li>
                            </ul>

                        </nav>
                    </div>
                    <div className="col-md-8 content">
                        {
                            (show === 0) ? (
                                <TableUsers />
                            ) :
                                (show === 1) ? (
                                    <TableRepresentatives />
                                ) :
                                    (show === 2) ? (
                                        <TableMessages />
                                        ) :
                                        (show === 3) ? (
                                        <TableVerifiedRep />) : 
                                        (show===4)? <AdminPendingActivities /> : <AdminCompletedActivities/>
                        }
                    </div>
                </div>
            </IconContext.Provider>
        </div> : (<Redirect to='/admin/login' />)
    )
}

export default SideBar