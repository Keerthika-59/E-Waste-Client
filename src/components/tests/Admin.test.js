import React from "react";
import ReactDOM from "react-dom";
import { Route } from "react-router-dom";
import AdminCompletedActivities from "../Admin/AdminCompletedActivities";
import AdminPendingActivities from "../Admin/AdminPendingActivities";
import RepCompletedActivities from "../Admin/RepCompletedActivities";
import SideBar from "../Admin/SideBar";
import TableMessages from "../Admin/TableMessages";
import TableRepresentatives from "../Admin/TableRepresentatives";
import TableUsers from "../Admin/TableUsers";
import TableVerifiedRep from "../Admin/TableVerifiedRep";
import UserCompletedActivities from "../Admin/UserCompletedActivities";

test("renders all completed activities", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <AdminCompletedActivities />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
test("renders all pending activities", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <AdminPendingActivities />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
test("renders representative completed activities", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <RepCompletedActivities />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});

test("renders sidebar", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <SideBar />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
test("renders table messages", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <TableMessages />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
test("renders representatives", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <TableRepresentatives />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
test("renders users", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <TableUsers />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
test("renders verified representatives", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <TableVerifiedRep />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
test("renders suser completed activities", () => {
  const div = document.createElement("div");
  <Route>
    ReactDom.render(
    <UserCompletedActivities />
    ,div);
  </Route>;
  ReactDOM.unmountComponentAtNode(div);
});
