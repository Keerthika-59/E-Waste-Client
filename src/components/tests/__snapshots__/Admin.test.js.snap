// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test AdminCompletedActivities component 1`] = `
<Fragment>
  <div
    className="row w-100"
  >
    <div
      className="col mb-3 col-12 text-center"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <PaginationComponent
            currentPage={1}
            itemsPerPage={9}
            onPageChange={[Function]}
            total={0}
          />
        </div>
      </div>
      <h4>
        Completed Activities
      </h4>
      <ForwardRef
        responsive={true}
      >
        <thead>
          <tr>
            <th>
              Activity Id
            </th>
            <th>
              User Name
            </th>
            <th>
              Bio Waste
            </th>
            <th>
              Non-Bio Waste
            </th>
            <th>
              Donation
            </th>
            <th>
              Representative Name
            </th>
          </tr>
        </thead>
        <tbody />
      </ForwardRef>
    </div>
  </div>
</Fragment>
`;

exports[`should test AdminPendingActivities component 1`] = `
<Fragment>
  <div
    className="row w-100"
  >
    <div
      className="col mb-3 col-12 text-center"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <PaginationComponent
            currentPage={1}
            itemsPerPage={9}
            onPageChange={[Function]}
            total={0}
          />
        </div>
      </div>
      <h4>
        Pending Activities
      </h4>
      <ForwardRef
        responsive={true}
      >
        <thead>
          <tr>
            <th>
              Activity Id
            </th>
            <th>
              User Name
            </th>
            <th>
              Bio Waste
            </th>
            <th>
              Non-Bio Waste
            </th>
            <th>
              Donation
            </th>
            <th>
              Representative Name
            </th>
          </tr>
        </thead>
        <tbody />
      </ForwardRef>
    </div>
  </div>
</Fragment>
`;

exports[`should test RepCompletedActivities component 1`] = `
<Fragment>
  <ForwardRef
    bordered={true}
    hover={true}
    responsive={true}
    striped={true}
  >
    <thead>
      <tr>
        <th>
          Activity Id
        </th>
        <th>
          Bio Waste
        </th>
        <th>
          Non Bio Waste
        </th>
        <th>
          Donation
        </th>
        <th>
          User Id
        </th>
      </tr>
    </thead>
    <tbody />
  </ForwardRef>
</Fragment>
`;

exports[`should test SideBar component 1`] = `
<Redirect
  to="/admin/login"
/>
`;

exports[`should test TableMessages component 1`] = `
<Fragment>
  <div
    className="row w-100"
  >
    <div
      className="col mb-3 col-12 text-center"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <PaginationComponent
            currentPage={1}
            itemsPerPage={15}
            onPageChange={[Function]}
            total={0}
          />
        </div>
      </div>
      <h4>
        Messages
      </h4>
      <ForwardRef
        responsive={true}
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              E-mail
            </th>
            <th>
              Message
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody />
      </ForwardRef>
    </div>
  </div>
</Fragment>
`;

exports[`should test TableRepresentatives component 1`] = `
<Fragment>
  <div
    className="row w-100"
  >
    <div
      className="col mb-3 col-12 text-center"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <PaginationComponent
            currentPage={1}
            itemsPerPage={9}
            onPageChange={[Function]}
            total={0}
          />
        </div>
      </div>
      <h4>
        Representative
      </h4>
      <ForwardRef
        responsive={true}
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              E-mail
            </th>
            <th>
              Phone Number
            </th>
            <th>
              City
            </th>
            <th>
              Address
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody />
      </ForwardRef>
    </div>
  </div>
</Fragment>
`;

exports[`should test TableUsers component 1`] = `
<Fragment>
  <div
    className="row w-100"
  >
    <div
      className="col mb-3 col-12 text-center"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <PaginationComponent
            currentPage={1}
            itemsPerPage={9}
            onPageChange={[Function]}
            total={0}
          />
        </div>
      </div>
      <h4>
        Users
      </h4>
      <ForwardRef
        responsive={true}
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              E-mail
            </th>
            <th>
              Phone Number
            </th>
            <th>
              City
            </th>
            <th>
              Address
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody />
      </ForwardRef>
    </div>
  </div>
</Fragment>
`;

exports[`should test TableVerifiedRep component 1`] = `
<Fragment>
  <div
    className="row w-100"
  >
    <div
      className="col mb-3 col-12 text-center"
    >
      <h3>
        Verify representatives
      </h3>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <PaginationComponent
            currentPage={1}
            itemsPerPage={7}
            onPageChange={[Function]}
            total={0}
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`should test UserCompletedActivities component 1`] = `
<Fragment>
  <ForwardRef
    bordered={true}
    hover={true}
    responsive={true}
    striped={true}
  >
    <thead>
      <tr>
        <th>
          Activity Id
        </th>
        <th>
          Bio Waste
        </th>
        <th>
          Non Bio Waste
        </th>
        <th>
          Donation
        </th>
        <th>
          Representative Id
        </th>
      </tr>
    </thead>
    <tbody />
  </ForwardRef>
</Fragment>
`;
